src/pybind/mgr/influx/module.py

```diff
@@ -219,11 +220,61 @@ def init_module_config(self):
             self.get_config("verify_ssl", default=self.config_keys['verify_ssl'])
         self.config['verify_ssl'] = verify_ssl.lower() == 'true'
 
-    def send_to_influx(self):
-        if not self.config['hostname']:
-            self.log.error("No Influx server configured, please set one using: "
-                           "ceph influx config-set hostname <hostname>")
+        # get_config_json returns None if key is not set, does not accept default arg 
+        self.config['destinations'] = \
+                self.get_config_json("destinations")
+
+        self.init_influx_clients()
+
+    def init_influx_clients(self):
+
+        self.clients = []
+        
+        if not self.config['destinations']:
+            destinations = [ 
+                { 
+                    'hostname':   self.config['hostname'],
+                    'username':   self.config['username'],
+                    'password':   self.config['password'],
+                    'database':   self.config['database'],
+                    'ssl':        self.config['ssl'],
+                    'verify_ssl': self.config['verify_ssl']
+                },
+            ]
+        else: 
+            destinations = self.config['destinations']
```

Comment:
This shouldn't ignore the `hostname` field just because destinations is set -- need to do something a bit friendlier here, like rolling the `hostname` etc fields into `destinations`, but checking for duplicates